internal/provider/peer_resource.go

```diff
@@ -0,0 +1,432 @@
+			return
+		}
+	}
+
+	resp.Diagnostics.Append(peerAPIToTerraform(ctx, peer, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Peer) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data PeerModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	peer, err := r.client.Peers.Get(ctx, data.Id.ValueString())
+
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			data.Id = types.StringNull()
+			resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+		} else {
+			resp.Diagnostics.AddError("Error getting Peer", err.Error())
+		}
+		return
+	}
+
+	resp.Diagnostics.Append(peerAPIToTerraform(ctx, peer, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Peer) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data PeerModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Id.ValueString() == "" {
+		r.Create(ctx, resource.CreateRequest{Config: req.Config, Plan: req.Plan, ProviderMeta: req.Config}, (*resource.CreateResponse)(resp))
+		return
+	}
+
+	peer, err := r.client.Peers.Update(ctx, data.Id.ValueString(), api.PeerRequest{
+		ApprovalRequired:            data.ApprovalRequired.ValueBoolPointer(),
+		InactivityExpirationEnabled: data.InactivityExpirationEnabled.ValueBool(),
+		LoginExpirationEnabled:      data.LoginExpirationEnabled.ValueBool(),
+		Name:                        data.Name.ValueString(),
+		SshEnabled:                  data.SshEnabled.ValueBool(),
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating Peer", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(peerAPIToTerraform(ctx, peer, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If applicable, this is a great opportunity to initialize any necessary
+	// provider client data and make a call using it.
+	// httpResp, err := r.client.Do(httpReq)
+	// if err != nil {
+	//     resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update peer, got error: %s", err))
+	//     return
+	// }
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Peer) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data PeerModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := r.client.Peers.Delete(ctx, data.Id.ValueString())
```

Comment:
Since it is managing an existing peer, we should probably not delete it unless explicitly told to with a separate parameter.

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7